tests/conftest.py

```diff
@@ -1015,3 +1048,103 @@ async def mocked_compiler_service_block(server, monkeypatch):
     monkey_patch_compiler_service(monkeypatch, server, True, runner_queue)
 
     yield runner_queue
+
+
+@pytest.fixture
+def tmpvenv(tmpdir: py.path.local) -> Iterator[Tuple[py.path.local, py.path.local]]:
+    """
+    Creates a venv with the latest pip in `${tmpdir}/.venv` where `${tmpdir}` is the directory returned by the
+    `tmpdir` fixture. The venv is activated within the currently running process. This venv is completely decoupled from the
+    active development venv. As a result, any attempts to load new modules from the development venv will fail until cleanup.
+
+    :return: A tuple of the paths to the venv and the Python executable respectively.
+    """
+    venv_dir: py.path.local = tmpdir.join(".venv")
+    python_path: py.path.local = venv_dir.join("bin", "python")
+    venv.create(venv_dir, with_pip=True)
```

Comment:
Who much time do we add to test execution by creating this many venv's?

I wonder if we could do this more efficiently (but this is of secondary importance)

Reply:
I assume this doesn't affect the total runtime too much since it's only used by a handful of tests. In any case, the `snippetcompiler` fixture already creates a new compiler venv directory for each test that uses it so I believe the performance impact of this will be negligible. I haven't tested it yet though, I'll make sure to do that to be certain.